subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/CompositeBuildTaskExecutionIntegrationTest.groovy

```diff
@@ -373,4 +390,39 @@ class CompositeBuildTaskExecutionIntegrationTest extends AbstractIntegrationSpec
         succeeds("doSomething")
         outputContains("do something")
     }
+
+    private def setupComplexExample() {
+        settingsFile << """
+            rootProject.name = 'root'
+            include('sub')
+            includeBuild('included')
+        """
+        file('included/settings.gradle') << """
+            include('sub')
+        """
+        buildFile << """
+            def test = tasks.register('test')
```

Comment:
These tasks should probably have an action or else the tests will always report up-to-date which can be a little confusing when things go wrong.

Reply:
Fixed